File: freecodecamp/2026-02-01.py
from datetime import datetime

def digital_detox(logs):
    # Parse and sort logs
    times = sorted(datetime.strptime(log, "%Y-%m-%d %H:%M:%S") for log in logs)

    daily_count = {}
    four_hours = 4 * 60 * 60  # seconds

    for i in range(len(times)):
        # Rule 1: 4-hour gap
        if i > 0:
            diff = (times[i] - times[i - 1]).total_seconds()
            if diff < four_hours:
                return False

        # Rule 2: max 2 per day
        day = times[i].date()
        daily_count[day] = daily_count.get(day, 0) + 1

        if daily_count[day] > 2:
            return False

    return True

File: freecodecamp/2026-02-01.js
function digitalDetox(logs) {
  // Convert to Date objects and sort
  const times = logs
    .map(log => new Date(log.replace(" ", "T")))
    .sort((a, b) => a - b);

  const FOUR_HOURS = 4 * 60 * 60 * 1000;
  const dailyCount = {};

  for (let i = 0; i < times.length; i++) {
    // Rule 1: 4-hour gap
    if (i > 0 && times[i] - times[i - 1] < FOUR_HOURS) {
      return false;
    }

    // Rule 2: max 2 per day
    const day = times[i].toISOString().split("T")[0];
    dailyCount[day] = (dailyCount[day] || 0) + 1;

    if (dailyCount[day] > 2) {
      return false;
    }
  }

  return true;
}
